geoh5py/objects/object_base.py

```diff
@@ -305,9 +164,127 @@ def add_data(
 
         return data_objects
 
+    def add_data_to_group(
+        self, data: Union[List, Data, uuid.UUID, str], name: str
+    ) -> PropertyGroup:
+        """
+        Append data children to a :obj:`~geoh5py.groups.property_group.PropertyGroup`
+        All given data must be children of the parent object.
+
+        :param data: :obj:`~geoh5py.data.data.Data` object,
+            :obj:`~geoh5py.shared.entity.Entity.uid` or
+            :obj:`~geoh5py.shared.entity.Entity.name` of data
+        :param name: Name of a :obj:`~geoh5py.groups.property_group.PropertyGroup`.
+            A new group is created if none exist with the given name.
+
+        :return property_group: The target property_group
+        """
+        prop_group = self.find_or_create_property_group(name=name)
+
+        children_uid = [child.uid for child in self.children]
+
+        def reference_to_uid(value):
+            if isinstance(value, Data):
+                uid = [value.uid]
+            elif isinstance(value, str):
+                uid = [
+                    obj.uid
+                    for obj in self.workspace.get_entity(value)
+                    if obj.uid in children_uid
+                ]
+            elif isinstance(value, uuid.UUID):
+                uid = [value]
+            return uid
+
+        if isinstance(data, list):
+            uid = []
+            for datum in data:
+                uid += reference_to_uid(datum)
+        else:
+            uid = reference_to_uid(data)
+
+        for i in uid:
+            assert i in [
+                child.uid for child in self.children
+            ], f"Given data with uuid {i} does not match any known children"
+
+        prop_group.properties = uid
+        self.modified_attributes = "property_groups"
+
+        return prop_group
+
+    @property
+    def cells(self):
+        """
+        :obj:`numpy.array` of :obj:`int`: Array of indices
+        defining the connection between
+        :obj:`~geoh5py.objects.object_base.ObjectBase.vertices`.
+        """
+        ...
+
+    @property
+    def comments(self):
+        """
+        Fetch a :obj:`~geoh5py.data.text_data.CommentsData` entity from children.
```

Comment:
"Fetch a ..."
does it mean there could be several but this just returns the first one?